application/src/main/java/org/opentripplanner/apis/transmodel/mapping/TripRequestMapper.java

```diff
@@ -25,83 +21,81 @@ public class TripRequestMapper {
-
     callWith.argument("from", (Map<String, Object> v) ->
-      request.setFrom(GenericLocationMapper.toGenericLocation(v))
+      requestBuilder.withFrom(GenericLocationMapper.toGenericLocation(v))
     );
     callWith.argument("to", (Map<String, Object> v) ->
-      request.setTo(GenericLocationMapper.toGenericLocation(v))
+      requestBuilder.withTo(GenericLocationMapper.toGenericLocation(v))
     );
     callWith.argument("passThroughPoints", (List<Map<String, Object>> v) -> {
-      request.setViaLocations(TripViaLocationMapper.toLegacyPassThroughLocations(v));
+      requestBuilder.withViaLocations(TripViaLocationMapper.toLegacyPassThroughLocations(v));
     });
     callWith.argument(TripQuery.TRIP_VIA_PARAMETER, (List<Map<String, Object>> v) -> {
-      request.setViaLocations(TripViaLocationMapper.mapToViaLocations(v));
+      requestBuilder.withViaLocations(TripViaLocationMapper.mapToViaLocations(v));
     });
 
     callWith.argument("dateTime", millisSinceEpoch ->
-      request.setDateTime(Instant.ofEpochMilli((long) millisSinceEpoch))
+      requestBuilder.withDateTime(Instant.ofEpochMilli((long) millisSinceEpoch))
     );
 
     callWith.argument("bookingTime", millisSinceEpoch ->
-      request.setBookingTime(Instant.ofEpochMilli((long) millisSinceEpoch))
-    );
-
-    callWith.argument("searchWindow", (Integer m) -> request.setSearchWindow(Duration.ofMinutes(m))
-    );
-    callWith.argument("pageCursor", request::setPageCursorFromEncoded);
-    callWith.argument("timetableView", request::setTimetableView);
-    callWith.argument("wheelchairAccessible", request::setWheelchair);
-    callWith.argument("numTripPatterns", request::setNumItineraries);
-    callWith.argument("arriveBy", request::setArriveBy);
-
-    callWith.argument("preferred.authorities", (Collection<String> authorities) ->
-      request.journey().transit().setPreferredAgencies(mapIDsToDomainNullSafe(authorities))
-    );
-    callWith.argument("unpreferred.authorities", (Collection<String> authorities) ->
-      request.journey().transit().setUnpreferredAgencies(mapIDsToDomainNullSafe(authorities))
+      requestBuilder.withBookingTime(Instant.ofEpochMilli((long) millisSinceEpoch))
     );
 
-    callWith.argument("preferred.lines", (List<String> lines) ->
-      request.journey().transit().setPreferredRoutes(mapIDsToDomainNullSafe(lines))
+    callWith.argument("searchWindow", (Integer m) ->
+      requestBuilder.withSearchWindow(Duration.ofMinutes(m))
     );
-    callWith.argument("unpreferred.lines", (List<String> lines) ->
-      request.journey().transit().setUnpreferredRoutes(mapIDsToDomainNullSafe(lines))
-    );
-
-    if (GqlUtil.hasArgument(environment, "modes")) {
-      request
-        .journey()
-        .setModes(RequestModesMapper.mapRequestModes(environment.getArgument("modes")));
-    }
+    callWith.argument("pageCursor", requestBuilder::withPageCursorFromEncoded);
```

Comment:
I just hope there is good test coverage for this.

Reply:
I don´t think there is, but we will test this PR in the Entur TEST environment instead with our request replayer. If will catch exceptions, but not "silent" bugs.